tesseract/tesseract_planning/tesseract_motion_planners/src/hybrid/ompl_trajopt_freespace_planner.cpp

```diff
@@ -120,6 +120,14 @@ tesseract_common::StatusCode OMPLTrajOptFreespacePlanner::solve(PlannerResponse&
     return response.status;
   }
 
+  if (ompl_planning_response.joint_trajectory.trajectory.rows() == 2)
```

Comment:
When you ran into this was your number of output states set to 2? If not then there may be an issue in the logic above as it should leverage OMPL interpolate to generate the set number of output states. This may be an issue in OMPL and if so we should interpolate using the interpolate function in the core folder utils.h for this conditional.

```
inline Eigen::MatrixXd interpolate(const Eigen::VectorXd& start,
                                   const Eigen::VectorXd& stop,
                                   int steps)
{
  assert(start.size() == stop.size());

  Eigen::MatrixXd result(start.size(), steps + 1);

  for (int i = 0; i < start.size(); ++i)
    result.row(i) = Eigen::VectorXd::LinSpaced(steps + 1, start(i), stop(i));

  return result;
}
```

Reply:
If you turn on OMPL smoothing, it can get to 2 states when there are no obstacles nearby.  I'm fine replacing my functionality with the one above if you like, though I'm unsure that it will change the final results.

Reply:
I looked at the tesseract ompl config doxygen and it does state that if simplify is set to true the number of output states is ignored. I think what you done is good. Though when using ompl + trajopt for I would recommend setting at minimum 10 to 20 states and simplify to false. In the case above trajopt will only have 3 states to adjust and you may have issues on the trajopt side if velocity, acceleration or jerk smoothing is enable because it is using finite difference to calculate these and there will not be enough states to do so.

Reply:
Well, if you interpolate as you indicated, that may be better.  The only issue would be if someone called the planner from a location they were already at for some reason.  Then you'd likely get 2 equal states that go to 3 equal states.  If TrajOpt can handle that, then great.

Reply:
> The only issue would be if someone called the planner from a location they were already at for some reason. Then you'd likely get 2 equal states that go to 3 equal states. If TrajOpt can handle that, then great.

Sorry, I am not following.

Reply:
If a person calls the planner and the start/goal are the same (let' say this is programmatic and maybe they didn't mean to, but it happened).  Then (given you are allowing the simplify=true option) OMPL is going to return two states, that are identical.  So even if you interpolate and/or just copy the last state to make a 3 state plan, they will all be the same joint angles.  

My comment on this was that if TrajOpt can handle this without blowing up (diving by zero or something), then it should be fine.  So, I'm inclined to like your interpolation solution better if you think that handles the more general case as well as this case.

Reply:
Now I understand. Should we maybe add a check early in the pipeline to catch this case and automatically return the input in the return response to prevent an unnecessary plan?

Reply:
You could.  But then there's always going to be some corner case right.  I'd say as long as TrajOpt is numerically stable in the 3 point case (even if they are all the same 3 points), then we use your interpolation change above and let it pass through.  My two cents.

Reply:
Sounds good.